initrd/etc/functions

```diff
@@ -19,18 +19,28 @@ mask_param() {
 # N=0 is the name of the command to be executed, N=1 is its first parameter,
 # etc.
 DO_WITH_DEBUG() {
-	if [ "$1" == "--mask-position" ]; then
-		mask_position="$2"
+	local exit_status
+	local cmd_output
+	DEBUG "PATH: $PATH"
+	local cmd=("$@")
+	if [[ "$1" == "--mask-position" ]]; then
+		local mask_position="$2"
 		shift
 		shift
-		DEBUG_ARGS=("$@")
-
-		DEBUG_ARGS[$mask_position]="$(mask_param "${DEBUG_ARGS[$mask_position]}")"
-		DEBUG "${DEBUG_ARGS[@]}"
-	else
-		DEBUG "$@"
+		cmd=("$@")
+		cmd[$mask_position]="$(mask_param "${cmd[$mask_position]}")"
+	fi
+	if [[ ${#cmd[@]} -eq 1 ]]; then
+		# If there's only one argument, try to split it into multiple arguments
+		read -a cmd <<< "${cmd[0]}"
 	fi
-	"$@"
+	DEBUG "Executing command with cmd: ${cmd[*]}"
+	# Sanitize the command output by removing special characters
+	cmd_output=$("${cmd[@]}" 2>&1 | sed 's/[&;|`$(){}<>]//g')
+	exit_status=$?
+	DEBUG "Command output: $cmd_output"
+	DEBUG "Command exited with status: $exit_status"
+	return $exit_status
```

Comment:
This is getting really noisy and now prevents you from capturing the output of the command when DO_WITH_DEBUG is used.  I think it will also now break with `set -e` if the command fails but the caller intended to check it.

IMO, we should break this out of this PR that says it is about some board-specific changes.  There is definitely merit to showing command failures, etc., but I think we should do it outside of a board-specific PR so we can discuss the impact properly.  The impact isvisible in this PR IMO since you now have to start duplicating what DO_WITH_DEBUG used to do everywhere, as it's too spammy with the changes.